lib/core/include/irods/library_features.h

```diff
@@ -1,6 +1,8 @@
 #ifndef IRODS_LIBRARY_FEATURES_H
 #define IRODS_LIBRARY_FEATURES_H
 
+// NOLINTBEGIN
```

Comment:
Is this intentional? Feels like there should be an end, or a comment explaining that this is intentional.

Reply:
Yes. This is intentional.

Clang-Tidy needs to ignore this file entirely because it is meant to be used for C and C++. We can't use anything other than macros unless we're going to require devs to link a shared library.

The `// NOLINTEND` is necessary, but I don't mind adding it if that helps.

Reply:
Okay, I figured as much. Just wasn't sure if it was a mistake that there was no end. I think it would be helpful to add the `NOLINTEND` or an explanatory comment to the effect of what you've said here. Not required, of course, since it works.

Reply:
Added words above clang-tidy directive.